src/components/Table/GridTable.tsx

```diff
@@ -195,21 +196,25 @@ export function GridTable<R extends Kinded, S = {}, X extends Only<GridTableXss,
     api.init(persistCollapse, virtuosoRef, rows);
     api.setActiveRowId(activeRowId);
     api.setActiveCellId(activeCellId);
+    api.tableState.initSortState(props.sorting, columns);
     return api;
   }, [props.api]);
 
   const style = resolveStyles(maybeStyle);
+  const { tableState } = api;
+  const { sortConfig, setSortKey } = tableState;
+  const { on: sortOn } = sortConfig ?? ({} as GridSortConfig);
```

Comment:
Thought: what does the type of `sortOn` end up being with the `as GridSortConfig`? I think the `as` means that `sortOn` could be `undefined` but it may be getting inferred as something else, like only `client | server`.

Reply:
Good point. It does come back as expecting it to be `client | server`. Which isn't correct, it should be allowed to be `undefined`. Will update. Thank for catching.